velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -284,16 +289,45 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
       if (isFileColumnNamesReadAsLowerCase()) {
         folly::toLowerAscii(childName);
       }
-      auto childRequestedType =
-          requestedType ? requestedType->asRow().findChild(childName) : nullptr;
+
+      TypePtr childRequestedType = nullptr;
+      if (requestedType && requestedType->isRow()) {
+        auto fileTypeIdx =
+            requestedType->asRow().getChildIdxIfExists(childName);
+        if (fileTypeIdx.has_value()) {
+          childRequestedType = requestedType->asRow().childAt(*fileTypeIdx);
+        }
+      }
+
+      // Handling elements of ARRAY/MAP
+      if (!requestedType && parentRequestedType) {
+        if (parentRequestedType->isArray()) {
+          childRequestedType = parentRequestedType->asArray().elementType();
+        } else if (parentRequestedType->isMap()) {
+          auto mapType = parentRequestedType->asMap();
+          auto isMapKey = schemaElement.__isset.repetition_type &&
+              schemaElement.repetition_type ==
+                  thrift::FieldRepetitionType::REPEATED &&
+              schema[parentSchemaIdx].converted_type ==
```

Comment:
@wypb Isn't this check a bit strict compared to the Map checks below?
converted type could be MAP_KEY_VALUE as well.
won't just `i == 0` work?

Reply:
@majetideepak Your suggestion is correct, `i== 0` is enough, because when running here, it must be processing the map type.